src/test/java/hudson/plugins/jira/JiraProjectPropertyTest.java

```diff
@@ -0,0 +1,131 @@
+        jiraFolderProperty.setSites(firstList);
+        folder.getProperties().add(jiraFolderProperty);
+    }
+
+    @Test
+    public void getSitesNullWithoutFolder() throws Exception {
+        FreeStyleProject freeStyleProject = r.createFreeStyleProject();
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        freeStyleProject.addProperty(prop);
+        JiraSite site = prop.getSite();
+        assertNull(site);
+        JiraProjectProperty actual = freeStyleProject.getProperty(JiraProjectProperty.class);
+        assertNotNull(actual);
+        assertNull(actual.getSite());
+    }
+
+    @Test
+    public void getSitesNullWithFolder() throws Exception {
+        freeStyleProject = folder.createProject(FreeStyleProject.class, "something");
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        freeStyleProject.addProperty(prop);
+        JiraProjectProperty property = freeStyleProject.getProperty(JiraProjectProperty.class);
+        assertNotNull(property);
+        assertNull(property.getSite());
+    }
+
+    @Test
+    public void testNoInitialDefault() throws Exception {
+        JiraProjectProperty prop = new JiraProjectProperty(null);
+        setupGlobalConfig("single-site.yml");
+        JiraSite site = prop.getSite();
+        assertNotNull(site);
+        assertEquals("https://jira.com/", site.getName());
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        freeStyleProject = r.createFreeStyleProject();
+        testProperty("single-site.yml", null);
+    }
+
+    @Test
+    public void testDefaultWithTwoEntries() throws Exception {
+        freeStyleProject = r.createFreeStyleProject();
+        testProperty("multiple-sites.yml", null);
+    }
+
+    @Test
+    public void testChooseWithOutFolder() throws Exception {
+        freeStyleProject = r.createFreeStyleProject();
+        JiraSite jiraSite = new JiraSite("https://jira.com/");
+        testProperty("multiple-sites.yml", jiraSite);
+    }
+
+    @Test
+    public void testChooseWithFolder() throws Exception {
+        freeStyleProject = folder.createProject(FreeStyleProject.class, "something");
+        testProperty("single-site.yml", firstList.get(0));
+    }
+
+    @Test
+    public void configRoundtripWithNestedFolder() throws Exception {
+        r.configRoundtrip(folder);
+        Folder secondFolder = folder.createProject(Folder.class, "second");
+        r.configRoundtrip(secondFolder);
+        freeStyleProject = secondFolder.createProject(FreeStyleProject.class, "something");
+        // testing we can get value from folder above.
+        testProperty("single-site.yml", firstList.get(0), true);
+    }
+
+    private void testProperty(String input, JiraSite expected) throws Exception {
+        testProperty(input, expected, false);
+    }
+
+     private void testProperty(String input, JiraSite expected, boolean roundTrip) throws Exception {
```

Comment:
could you please get rid of that method?
1. the `roundTrip` property is really used only in one test, it would be better to move that part to that test
2. conditional logic here complicates reading and understanding of what is test doing
3. you have to jump back and forth through the code to understand each test

in overall it's better to copy-paste few lines and have stupidly straightforward test cases rather then making your tests as complex as your code